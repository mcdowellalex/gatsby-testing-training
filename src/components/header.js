import React from "react"
import { StaticQuery, graphql } from "gatsby"

// testable component
// takes in data prop instead of using a static query
export const PureHeader = ({ data }) => (
  <header>
    <h1>{data.site.siteMetadata.title}</h1>
  </header>
)

// defualt export component used in the application
export const Header = props => (
  <StaticQuery
    query={graphql`
      query {
        site {
          siteMetadata {
            title
          }
        }
      }
    `}
    render={data => <PureHeader {...props} data={data} />}
  />
)

export default Header

// Now you have two components exported from the file: the component that 
// includes the StaticQuery data which is still the default export, and another 
// component that you can test. This means you can test the component independently of 
// the GraphQL. In addition, whether you utilize StaticQuery or useStaticQuery, your test 
// should still function properly.
