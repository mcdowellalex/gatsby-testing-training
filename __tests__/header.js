// // using jest to do snapshot tests 

// import React from "react"
// import renderer from "react-test-renderer"

// import Header from "../header"

// describe("Header", () => {
//   it("renders correctly", () => {
//     const tree = renderer
//       .create(<Header siteTitle="Default Starter" />)
//       .toJSON()
//     expect(tree).toMatchSnapshot()
//   })
// })


//testing graphQL stuff
import React from "react"
import renderer from "react-test-renderer"

import { PureHeader as Header } from "../src/components/header"

describe("Header", () => {
  it("renders correctly", () => {
    // Created using the query from Header.js
    const data = {
      site: {
        siteMetadata: {
          title: "Gatsby Starter Blog",
        },
      },
    }
    const tree = renderer.create(<Header data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})