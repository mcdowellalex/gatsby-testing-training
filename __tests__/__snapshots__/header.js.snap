// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <h1>
    Gatsby Starter Blog
  </h1>
</header>
`;
